#Binds ppc to use only one core to workaround threading problems for ppc.
http://cvs.fedoraproject.org/viewvc/rpms/mono/devel/mono-2.2-ppc-threading.patch?view=log

--- mono-2.2/mono/mini/main.c	2008-11-18 19:00:26.000000000 +0000
+++ mono-2.2/mono/mini/main-new.c	2008-11-18 19:16:11.000000000 +0000
@@ -26,10 +26,14 @@
 }
 
 #else
-
+#include <sched.h>
 int
 main (int argc, char* argv[])
 {
+#ifdef __powerpc__
+  unsigned long mask = 1;
+  sched_setaffinity(0, sizeof(mask), &mask);
+#endif
 	mono_build_date = build_date;
 	return mono_main (argc, argv);
 }
